src/Neo/SmartContract/Native/StdLib.cs

```diff
@@ -131,6 +131,35 @@ public static byte[] Base64Decode([MaxLength(MaxInputLength)] string s)
             return Convert.FromBase64String(s);
         }
 
+        /// <summary>
+        /// Encodes a byte array into a base64Url string.
+        /// </summary>
+        /// <param name="data">The byte array to be encoded.</param>
+        /// <returns>The encoded base64Url string.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 5)]
+        public static string Base64UrlEncode([MaxLength(MaxInputLength)] byte[] data)
+        {
+            string base64 = Convert.ToBase64String(data);
+            return base64.TrimEnd('=').Replace('+', '-').Replace('/', '_');
+        }
+
+        /// <summary>
+        /// Decodes a byte array from a base64Url string.
+        /// </summary>
+        /// <param name="s">The base64Url string.</param>
+        /// <returns>The decoded byte array.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 5)]
+        public static byte[] Base64UrlDecode([MaxLength(MaxInputLength)] string s)
+        {
+            var padded = s.Replace('-', '+').Replace('_', '/');
+            switch (padded.Length % 4)
+            {
+                case 2: padded += "=="; break;
+                case 3: padded += "="; break;
+            }
```

Comment:
What is this for? This is wrong.

 - Padding is skipped so the pad character '=' doesn't have to be percent encoded.
 - The 62nd and 63rd regular base64 encoding characters ('+' and '/') are replaced with ('-' and '_'). This makes the encoding alphabet URL safe.

Reply:
updated